web-admin/tests/public-urls.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { expect } from "@playwright/test";
+import { test } from "./setup/base";
+
+test.describe("Public URLs", () => {
+  let publicUrl: string;
+
+  test("should be able to create a public URL", async ({ page }) => {
```

Comment:
Should this instead be a beforeAll? Feels like an anti-pattern for other tests depend on this, but not enforced to be dependant.

Reply:
True. I want the "create a public URL" step to be an official "test", so I've marked the test suite as `serial`

Reply:
This still doesnt fully fix the issue. If someone adds a new test, they have to make sure to run this and the new test all the time.

I think we can still have a test that asserts the contents of `publicUrl` but creation can happen in `beforeAll`. It is cleaner than having to remember to run both tests manually.

Reply:
Found an issue talking about the exact thing: https://github.com/microsoft/playwright/issues/10114
Leading to https://playwright.dev/docs/api/class-test#test-step

Worth checking it out.

Reply:
In this case, the `test.step` approach doesn't really work because each test operates on one `page`. Here, we need one test for an `adminPage` and one test for an `anonPage`.

Reply:
I don't think there's a perfect solution. I added a `test.skip` note to make it a bit more clear that the `adminPage` and `anonPage` tests depend on the `publicURL` variable.